### PR TITLE
encode stringIO bodies to utf-8

### PR DESCRIPTION
**Pull request by zawan-ila**
Closes [3053](https://github.com/urllib3/urllib3/issues/3053)

I am not sure if I should add a warning.

### CODE REVIEW

File: test/with_dummyserver/test_socketlevel.py [hunk 1]
```diff
@@ -2380,7 +2380,7 @@ def body_generator() -> typing.Generator[bytes, None, None]:
             body.seek(0, 0)
             should_be_chunked = True
         elif body_type == "file_text":
-            body = io.StringIO("x" * 10)
+            body = io.StringIO("x\x80\x81")
```
Comment (pquentin) [inline test/with_dummyserver/test_socketlevel.py:2383]: We want all chunked bodies to be encoded to UTF-8, not only StringIO. Can you please use `x * 9 + \x80` everywhere?

**Review by pquentin — COMMENTED** [review]
Thanks! That's a good step in the right direction. We still need to make a few changes. While most of them are mentioned inline, another one is that we need a test for the encoding of `body` too, since we don't have one. See https://github.com/urllib3/urllib3/pull/3063/files for inspiration, where you can copy `test_encode_body_latin_1` (but rename it `test_encode_body_utf8` and use UTF-8).

File: docs/v2-migration-guide.rst [hunk 2]
```diff
@@ -55,6 +55,7 @@ Here's a short summary of which changes in urllib3 v2.0 are most important:
 - Changed the default minimum TLS version to TLS 1.2 (previously was TLS 1.0).
 - Removed support for verifying certificate hostnames via ``commonName``, now only ``subjectAltName`` is used.
 - Removed the default set of TLS ciphers, instead now urllib3 uses the list of ciphers configured by the system.
+- Changed the default encoding for string bodies to ``utf-8``.
```
[the remark this answers is not in the document]
Comment (zawan-ila) [reply docs/v2-migration-guide.rst:58]: `str body chunks` seems somewhat unclear. e.g Does it include string bodies sent as chunked encoding? I feel that using `StringIO bodies` or `string streams` would be clearer?

**Comment by pquentin** [comment]
This was ultimately done by @sethmlarson in https://github.com/urllib3/urllib3/pull/3428. Sorry for missing this pull request.

**Pull request closed without merging**